modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/RichTextArea.java

```diff
@@ -0,0 +1,2395 @@
+        public static final FunctionTag SELECT_TO_LINE_END = new FunctionTag();
+        /** Extends selection to the start of the visual text line at caret. */
+        public static final FunctionTag SELECT_TO_LINE_START = new FunctionTag();
+        /** Extends selection one visual text line up. */
+        public static final FunctionTag SELECT_UP = new FunctionTag();
+        /** Selects a word at the caret position. */
+        public static final FunctionTag SELECT_WORD = new FunctionTag();
+        /** Extends selection to the previous word (LTR) or next word (RTL). */
+        public static final FunctionTag SELECT_WORD_LEFT = new FunctionTag();
+        /** Extends selection to the beginning of next word. */
+        public static final FunctionTag SELECT_WORD_NEXT = new FunctionTag();
+        /** Extends selection to the end of next word. */
+        public static final FunctionTag SELECT_WORD_NEXT_END = new FunctionTag();
+        /** Extends selection to the previous word. */
+        public static final FunctionTag SELECT_WORD_PREVIOUS = new FunctionTag();
+        /** Extends selection to the next word (LTR) or previous word (RTL). */
+        public static final FunctionTag SELECT_WORD_RIGHT = new FunctionTag();
+        /** Inserts a tab symbol at the caret (editable), or transfer focus to the next focusable node. */
+        public static final FunctionTag INSERT_TAB = new FunctionTag();
+        /** If possible, undoes the last modification. */
+        public static final FunctionTag UNDO = new FunctionTag();
+
+        private Tags() { }
+    }
+
+    private SimpleObjectProperty<StyledTextModel> model;
+    private final SelectionModel selectionModel = new SingleSelectionModel();
+    private SimpleBooleanProperty editableProperty;
+    private SimpleObjectProperty<SideDecorator> leftDecorator;
+    private SimpleObjectProperty<SideDecorator> rightDecorator;
+    private ReadOnlyBooleanWrapper undoable;
+    private ReadOnlyBooleanWrapper redoable;
+    // styleables
+    private SimpleStyleableObjectProperty<Duration> caretBlinkPeriod;
+    private SimpleStyleableObjectProperty<Insets> contentPadding;
+    private StyleableBooleanProperty displayCaret;
+    private StyleableBooleanProperty highlightCurrentParagraph;
+    private StyleableBooleanProperty useContentHeight;
+    private StyleableBooleanProperty useContentWidth;
+    private StyleableBooleanProperty wrapText;
+    private RTAccessibilityHelper accessibilityHelper;
+    // will be moved to Control JDK-8314968
+    private final InputMap inputMap = new InputMap(this);
+
+    /** The style handler registry instance, made available for use by subclasses to add support for new style attributes. */
+    protected static final StyleHandlerRegistry styleHandlerRegistry = initStyleHandlerRegistry();
+
+    /**
+     * Creates the instance with the in-memory model {@link RichTextModel}.
+     */
```

Comment:
I recommend adding  `@defaultValue`

Reply:
Can you clarify please?  `@defaultValue` seem to apply to a property, this is a constructor.